controllers/persist/object/job/marsjob_persist_controller.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2020 The Alibaba Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package job
+
+import (
+	"context"
+
+	"github.com/alibaba/kubedl/apis/mars/v1alpha1"
+	"github.com/alibaba/kubedl/controllers/persist/util"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlruntime "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+func init() {
+	jobPersistCtrlMap[&v1alpha1.MarsJob{}] = NewMarsJobPersistController
+}
+
+func NewMarsJobPersistController(mgr ctrl.Manager, handler *jobPersistHandler) PersistController {
+	return &MarsJobPersistController{
+		client:  mgr.GetClient(),
+		handler: handler,
+	}
+}
+
+var _ reconcile.Reconciler = &MarsJobPersistController{}
+
+type MarsJobPersistController struct {
```

Comment:
looks like we can merge other job types' persistent controller into a single controller, and have the controller watch for all job's events.  
we can open an issue and do it later.

Reply:
I've tried this implementation at very beginning, we need to watch event sources from all jobs and enqueue them with their `Kind`, e.g: encode `Kind` info to `ctrl.Request` and decode it when we process this request, then we know which job to get by client.

another proposal is handling all persistency in job-event-handlers.